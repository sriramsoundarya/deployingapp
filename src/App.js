import logo from './logo.svg';
import './App.css';
import React ,{lazy,Suspense} from 'react'
import { Switch, Route,Redirect} from 'react-router-dom';
import Header from './components/Header'
import Content from './components/Content'
import Contact from './components/Contact';
import Dashboard from './components/Dashboard';
import Login from './components/Login';
import Projects from './components/Projects';
import Project from './components/Project';
import PrivateRoute from './components/PrivateRoute';
import Topics from './components/Topics';
const About=lazy(()=>import('./components/About'));

function App() {
  const location=window.location.pathname;
  console.log("location"+location);
  return (
    <div  className='App'>
      <Header/>
      <Suspense fallback={<h1>Loading...</h1>}>
      <Switch>
        <Route path="/about">
          <About/>
        </Route>
        <Route path="/Contact">
          <Contact/>
        </Route>
        <PrivateRoute path="/Dashboard">
          <Dashboard/>
        </PrivateRoute>
        <Route path="/Projects/:id">
          <Project/>
        </Route>
        <PrivateRoute path="/Projects">
          <Projects/>
        </PrivateRoute>
        <PrivateRoute path="/topics">
          <Topics/>
        </PrivateRoute>
        <Route path="/login">
          <Login/>
        </Route>
        <Route path="/logout">
          <Redirect to="/login"/>
        </Route>
        <Route path="*">
          <h1>404 error</h1>
        </Route>
      </Switch>
      </Suspense>
    </div>
  );
}

export default App;
