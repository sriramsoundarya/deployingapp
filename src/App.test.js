import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import App from './App'; // Update the import path accordingly

it('snapshot testing for App Component', () => {
  const tree = renderer
    .create(
      <Router>
        <App />
      </Router>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
