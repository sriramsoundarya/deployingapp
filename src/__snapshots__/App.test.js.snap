// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing for App Component 1`] = `
<div
  className="App"
>
  <nav
    className="navbar bg-dark justify-content-between"
  >
    <li
      className="nav-link"
    >
      <a
        aria-current={null}
        href="/gituser"
        onClick={[Function]}
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        Github Users
      </a>
    </li>
    <li
      className="nav-link"
    >
      <a
        aria-current={null}
        href="/Dashboard"
        onClick={[Function]}
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        Dashboard
      </a>
    </li>
    <li
      className="nav-link"
    >
      <a
        aria-current={null}
        href="/Projects"
        onClick={[Function]}
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        My Projects
      </a>
    </li>
    <li
      className="nav-link"
    >
      <a
        aria-current={null}
        href="/topics"
        onClick={[Function]}
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        My topics
      </a>
    </li>
    <li
      className="nav-link"
    >
      <a
        aria-current={null}
        href="/about"
        onClick={[Function]}
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        About
      </a>
    </li>
    <li
      className="nav-link"
    >
      <a
        aria-current={null}
        href="/Contact"
        onClick={[Function]}
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        Contact Us
      </a>
    </li>
    <li
      className="nav-link"
    >
      <a
        aria-current={null}
        href="/Login"
        onClick={[Function]}
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        Login
      </a>
    </li>
    <li
      className="nav-link"
    >
      <a
        aria-current={null}
        href="/Logout"
        onClick={[Function]}
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        Logout
      </a>
    </li>
  </nav>
  <h1>
    404 error
  </h1>
</div>
`;
